composition-js/src/merging/merge.ts

```diff
@@ -690,8 +695,20 @@ class Merger {
   private mergeObject(sources: (ObjectType | undefined)[], dest: ObjectType) {
     const isEntity = this.hintOnInconsistentEntity(sources, dest);
     const isValueType = !isEntity && !dest.isRootType();
+    const typeOverriddenInSubgraphs = sources
+      .map((src, idx) => {
+        if (src) {
+          const overrideDirectiveName = this.metadata(idx).overrideDirective().name;
+          const overrideDirective = src.appliedDirectives.find(d => d.name === overrideDirectiveName);
```

Comment:
Since the type of the source is `ObjectType`, I wasn't able to figure out how to do the `ofExtension` comparison for this version. Let me know what you think.